clients/gatekeeper.go

```diff
@@ -51,14 +51,44 @@ type (
 	}
 
 	Permission       map[string]interface{}
-	Permissions      map[string]Permission
+	Permissions      map[PermissionScope]Permission
 	UsersPermissions map[string]Permissions
 )
 
+// PermissionScope represents the specific permissions granted to a user.
+//
+// PermissionScope is a type alias rather than a user-defined type, because
+// the implementation of the gatekeeper client is inconsistent in its use of
+// UserPermissions and Permissions. Specifically, the UserInGroup function
+// returns Permissions, but it should be returning UserPermissions. As a
+// result, the Permissions type's keys have to include both strings (user ids)
+// in addition to PermissionScope.
+//
+// If a future update migrates UserInGroup to return UserPermissions, then
+// PermissionScope could be migrated to a user-defined type, and provide a
+// little more compile-time type checking.
+type PermissionScope = string
+
+const (
+	// PermFollow indicates that the user is permitted to follow another user
+	// via the Care Partner app.
+	PermFollow PermissionScope = "follow"
```

Comment:
[nit] Everywhere else it is "Permission" (non-abbreviated), but using "Perm" just here. Suggest you use non-abbreviated here, too.

Reply:
Sure, just thought it was still clear, and saved a few chars. I'll change it.